### PR TITLE
Installation MCP en 1 commande — install.sh + CLAUDE.md + section MCP dans le README

### PR DESCRIPTION
**Pull request by R1LGM**
## Ce que cette PR ajoute

Une couche d'installation pour les utilisateurs qui veulent récupérer les skills **et** brancher le MCP LGM dans Claude Code en une commande. Le repo expose déjà la voie skill-par-skill via `npx skills add` — cette PR ajoute le `curl | sh` qui fait tout d'un coup, plus un `CLAUDE.md` qui auto-onboarde les users qui clonent le repo.

---

### `install.sh` (nouveau, racine, exécutable)

Script POSIX shell qui :

1. Vérifie Node.js / npx.
2. Installe les 5 skills via `npx skills add LaGrowthMachine/gtm-system` (Claude Code, Cursor, Codex).
3. Demande si l'user a un compte LGM.
   - **Oui** → prompt pour l'API key (saisie masquée via `stty -echo`).
   - **Non** → affiche un pitch LGM (4 bénéfices) + un lien register UTM-taggé, puis skip la config MCP sans toucher à Claude.
4. Enregistre le MCP via la commande canonique Claude Code :
   ```
   claude mcp add --transport http LaGrowthMachine \
     https://mcpapp.lagrowthmachine.com/mcp \
     --header "Authorization:Bearer <key>"
   ```
   La config va dans `~/.claude.json` (géré par la CLI Claude Code), l'API key passe en Bearer token. Transport HTTP natif — **pas de `mcp-remote`, pas de dance OAuth, pas de browser**. Le MCP est connecté dès le 1er restart de Claude Code.
5. Affiche des prompts de démarrage + un soft nudge "star the repo". Si le MCP est skip, ajoute une ligne précisant que les skills marchent en standalone.

**Idempotent** : un re-run détecte l'entrée `LaGrowthMachine` via `claude mcp list` et skip proprement. Atteignable via :
```
curl -fsSL https://raw.githubusercontent.com/LaGrowthMachine/gtm-system/main/install.sh | sh
```
Alternative "review-before-run" documentée dans le README. Env vars `LGM_API_KEY` / `LGM_INSTALL_NO_CLAUDE_PERMS` / `LGM_INSTALL_NO_LAUNCH` supportées pour CI / tests.

### `CLAUDE.md` (nouveau, racine)

Auto-onboarding pour les users qui clonent le repo et l'ouvrent dans Claude Code. Détecte si le MCP est configuré via `claude mcp list` ; si non, propose de lancer `sh install.sh` (en précisant que l'install est interactive). Si oui, liste les outils MCP (Audiences / Campaigns / Conversations / Workspace) et les 5 skills, groupés par catégorie v2.

### `README.md` (mis à jour)

Nouvelle section "Install everything in one command" avec le curl one-liner — décrivant fidèlement le flow interactif et l'enregistrement via `claude mcp add` (plus de fausse promesse "wired into settings.json automatically"). Les voies existantes préservées sous "Other ways to install" (npx skills add pour une skill, manual clone + cp -r, Claude.ai web pour le MCP). Catalog v2, table MCP, Tally et liens LGM intacts.

### Réalignement de `won-deal-icp-finder` (rattrapage du rebuild v2)

Deux corrections qui avaient été perdues quand la skill a été re-uploadée sur main :

- **SKILL.md** : suppression du lien Markdown en chemin relatif vers le catalogue dans l'output rendu en chat (mention "GTM System catalog" en texte simple) — même traitement que les 4 autres skills, conformément à la règle "no relative paths".
- **README.md** : alignement de la section Install sur le format hybride (one-liner `npx` + fallback `git clone`) utilisé dans toute la librairie.

---

## Décisions clés

| Sujet | Choix |
|---|---|
| URL MCP | `https://mcpapp.lagrowthmachine.com/mcp` |
| Auth | API key en Bearer token (`Authorization:Bearer <key>`) — testé en direct contre le serveur, accepté |
| Méthode d'enregistrement | `claude mcp add --transport http` (canonique Claude Code, transport HTTP natif — comme Lemlist) |
| Stockage config | `~/.claude.json` géré par la CLI Claude Code (pas `settings.json` direct) |
| Alias | `LaGrowthMachine` → matche `mcp__LaGrowthMachine__*` que les 5 skills détectent, zéro refacto skill |
| Install interactive | Oui — l'user doit avoir un compte LGM + paste son API key |

---

## Tests end-to-end passés (local)

| Path | Statut |
|---|---|
| User existant, nominal flow | ✅ `claude mcp list` → Connected, "list my LGM campaigns" → 25/389 campagnes live |
| No account → pitch + register link, MCP skip propre | ✅ |
| API key vide → warn + skip, pas de config partielle | ✅ |
| Non-interactif via `LGM_API_KEY=...` | ✅ MCP enregistré sans prompt |
| Idempotence (re-run) | ✅ "already configured — skipping" |

---

## Pas touché

Les 4 skills `get-qualified-meetings` + `sales-nav-search-builder` sont intactes. Seul `won-deal-icp-finder` reçoit les 2 micro-fixes de réalignement décrits ci-dessus (cohérence cross-library).

→ **On ne merge pas — Brice valide.**

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+You are looking at the **GTM System repo** maintained by [La Growth Machine](https://lagrowthmachine.com). It bundles a library of public Claude skills (sourcing, list building, campaigns, copywriting, analytics) and an MCP server that lets Claude act directly inside the user's LGM workspace.
+
+## First-time setup
+
+Check if the LGM MCP is already configured by looking for `mcpServers.LaGrowthMachine` in `~/.claude/settings.json`.
+
+**If it is NOT configured**, proactively tell the user:
+
+> "I can see this is the LGM GTM Skills repo. Would you like me to install the LGM MCP and skills? It takes 30 seconds — I'll run `sh install.sh` for you."
```
Comment (bricemaurin) [inline CLAUDE.md]: 2 remarques
1. pour connecter le MCP il faut etre user (avoir une api key)
2. on devrait expliquer les bénéfices ici

si il est pas user le message devrait donner les bénéfices a l'utiliser, prévenir qu'on doit avoir un compte et proposer un lien pour en créer un (avc utm)

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+---
+
+### Layer 2 — GTM Skills (published in this repo)
+
+Skills guide Claude through complex GTM workflows. Just describe what you want. Skills are categorized into 4 outcomes mirroring the GTM motion:
+
+**Fuel my pipeline** — sourcing, list building, ICP
+
+- `sales-nav-search-builder` — turn a natural-language ICP into a precise LinkedIn Sales Navigator search URL, ready to import as an LGM audience
```
Comment (bricemaurin) [inline CLAUDE.md:63]: FYI : elise m'a demandé si ca marchait aussi en mode lookalike, et ca marche étonnement bien.

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+fi
+
+say()  { printf "%b\n" "$*"; }
+warn() { printf "%b\n" "${YELLOW}$*${RESET}" >&2; }
+fail() { printf "%b\n" "${RED}$*${RESET}" >&2; exit 1; }
+
+print_logo() {
+  if [ -t 1 ]; then
+    say "${CYAN}"
+    say "  ██╗      ██████╗ ███╗   ███╗"
```
Comment (bricemaurin) [inline install.sh:79]: j'adore l'ascii :)
est ce qu'on met LGM ou La Growth Machine ?

File: CLAUDE.md (hunk 2 above)
Comment (R1LGM) [reply CLAUDE.md:63]: ah trop bien ! j'avais pas tester mais c'est nickel

File: install.sh (hunk 3 above)
Comment (R1LGM) [reply install.sh:79]: Je pense que "La Growth Machine" ça va être énorme non ? Après on pourra ajuster la couleur et tout si besoin :)

**Comment by R1LGM** [comment]
Hello Brice — la PR est à jour et testée end-to-end. Récap complet.

### Tes 2 retours initiaux → intégrés

**1. "Pour connecter le MCP il faut être user (API key)"** → le script demande *"Do you already have an LGM account?"* et n'écrit rien dans la config tant qu'il n'a pas l'API key. Yes → prompt clé (saisie masquée `stty -echo`). No → pitch + skip propre.

**2. "Expliquer les bénéfices + lien register avec UTM"** → quand l'user n'a pas de compte :

```
La Growth Machine — what the MCP unlocks
  • Run multichannel outbound (LinkedIn, email, voice, calls) from one workspace
  • Pull live campaign data, replies and pipeline straight into Claude
  • Push Sales Navigator audiences into LGM with one prompt
  • Native HubSpot integration — every deal mapped back to the campaign that won it

  Create your free LGM account (14-day trial):
  https://app.lagrowthmachine.com/register/?utm_source=github&utm_medium=installer&utm_campaign=gtm-system-installer&utm_content=register
```

### Ton retour sur le drift → résolu

Ton Claude avait raison : `install.sh`, `CLAUDE.md` et `README.md` décrivaient 3 stratégies d'install différentes. Mon erreur — en basculant sur `claude mcp add` j'avais modifié seulement `install.sh`. Les 3 sont maintenant alignés sur **strategy (A) `claude mcp add --transport http`** (c'est ce que le code fait réellement) :

- `CLAUDE.md` : détection via `claude mcp list` (la config va dans `~/.claude.json` via la CLI, pas `settings.json`)
- `README.md` : plus de "wired into settings.json automatically" — on précise que l'install est **interactive**
- PR description réécrite (plus de refs à `mcp-remote` / `LGM_SETTINGS_OVERRIDE` qui n'existent plus)

### Bonus — réalignement `won-deal-icp-finder`

En scannant le repo j'ai trouvé 2 fixes qui avaient sauté quand la skill a été re-uploadée sur main : un relative path Markdown dans `SKILL.md:137` + l'install command du README en ancien format. Les deux réalignés sur les conventions de la librairie.

### Tests end-to-end (local)

| Path | Statut |
|---|---|
| User existant, nominal | ✅ `claude mcp list` → Connected, "list my LGM campaigns" → 25/389 live |
| No account → pitch + register link | ✅ skip propre |
| API key vide | ✅ warn + skip |
| Non-interactif `LGM_API_KEY=...` | ✅ enregistré sans prompt |
| Idempotence | ✅ "already configured — skipping" |

### Question ouverte

Le MCP accepte `Authorization: Bearer <api_key>` directement (testé en live contre `mcpapp.lagrowthmachine.com/mcp`). C'est le pattern long-terme ou un mode legacy ?

→ pas mergé, à toi de valider.

**Comment by bricemaurin** [comment]
👍 Le drift entre `install.sh`, `CLAUDE.md`, `README.md` et la description du PR est corrigé proprement — les 4 décrivent maintenant la même chose (`claude mcp add --transport http`, Bearer API key, `~/.claude.json`). Les 2 micro-fixes sur `won-deal-icp-finder` sont bien là aussi.

## 🟡 Un seul point à valider avant merge — scope du `claude mcp add`

[`install.sh:245`](https://github.com/LaGrowthMachine/gtm-system/blob/installation-mcp/install.sh#L245)

```sh
claude mcp add --transport http "$MCP_ALIAS" "$MCP_URL" \
     --header "Authorization:Bearer $LGM_API_KEY_VALUE"
```

Pas de flag `--scope`. Dans les versions récentes de Claude Code CLI, le scope par défaut est **`local`** (project-specific, écrit dans le `.mcp.json` du dossier courant) — **pas** `user` (global, dans `~/.claude.json` comme annoncé dans la description du PR et le README).

**Conséquence selon où l'user lance le `curl | sh`** :
- Depuis `~/` → atterrit en user scope, tout va bien (probablement ce qui a été testé en local).
- Depuis un dossier projet (ce que beaucoup de devs font instinctivement) → le MCP n'existe que pour ce projet. Et le check `claude mcp list` dans `CLAUDE.md` ne le retrouve plus une fois sorti du projet.

**Fix safe** : ajouter `--scope user` à la commande, et au fallback ligne 241 + au warning ligne 252 :

```sh
claude mcp add --scope user --transport http \"\$MCP_ALIAS\" \"\$MCP_URL\" \\
     --header \"Authorization:Bearer \$LGM_API_KEY_VALUE\"
```

## Polish non-bloquant

- `SCRIPT_VERSION=\"2026.05.27.7\"` (ligne 29) — daté d'avant le rebuild, à bumper.
- `>/dev/null 2>&1` (lignes 125, 246) mange les erreurs utiles. Une env var `LGM_INSTALL_VERBOSE=1` aiderait pour le debug en cas de support ticket.
- `Authorization:Bearer` sans espace (lignes 241, 246, 252) — non-standard HTTP mais le serveur LGM l'accepte (testé). À noter pour cohérence si un autre serveur strict apparaît un jour.

Le reste est OK pour merger une fois le `--scope user` ajouté.

**Comment by R1LGM** [comment]
Merci pour la review 🙏 — point bloquant + polish traités.

### 🟢 `--scope user` (bloquant) — corrigé

Tu avais raison, c'était un vrai bug. Sans `--scope user`, un `curl | sh` lancé depuis un dossier projet écrivait le MCP en local scope (`.mcp.json`) → invisible ailleurs. Ajouté `--scope user` partout :
- la commande réelle (ex-ligne 245)
- le fallback "claude CLI absent" (ex-241) + le warning d'échec (ex-252)
- le commentaire d'en-tête du script
- + aligné dans `README.md` et `CLAUDE.md` (pour rester sans drift)

**Re-testé spécifiquement ton scénario** : install depuis `~/Desktop/gtm-system`, puis `claude mcp list` depuis `~/` → `LaGrowthMachine: ... ✓ Connected`. Le MCP est bien global maintenant.

### Polish

- **Version bumpée** : `2026.05.27.7` → `2026.06.02.1`
- **`LGM_INSTALL_VERBOSE=1`** ajouté : un helper `run_quiet` silence `npx` / `claude mcp add` par défaut mais affiche tout l'output si la var est à 1 (pour les tickets support). Testé — on voit bien le détail des 5 skills + le `claude mcp add` complet.
- **`Authorization:Bearer` sans espace** : laissé tel quel puisque testé OK contre le serveur LGM. Noté pour cohérence si un serveur strict apparaît un jour.

### Couverture de test (end-to-end, local)

| Chemin | Statut |
|---|---|
| Nominal (env var) | ✅ user scope |
| Scope user visible depuis un autre dossier | ✅ (ton point) |
| Verbose | ✅ output complet |
| Idempotence (re-run) | ✅ skip propre |
| Live MCP call | ✅ 389 campagnes listées |
| **Interactif : `y` + API key masquée** | ✅ enregistré |
| **Interactif : `n` → pitch + skip propre** | ✅ |

→ Prêt à merger quand tu veux

image

**Comment by R1LGM** [comment]
⚠️ Un dernier bug attrapé avant merge — `curl | sh` était cassé (maintenant corrigé)

En testant le **vrai parcours de prod** (`curl ... | sh`, donc stdin = pipe) au lieu de `sh install.sh` (fichier local), le script s'arrêtait **juste après `npx skills add`** et ne posait jamais le prompt MCP.

**Cause** : en mode pipe, stdin = le flux du script lui-même. `npx skills add` lit stdin → il avalait toutes les lignes restantes du script (l'étape MCP comprise). Quand npx finissait, sh n'avait plus rien à lire → fin prématurée. Invisible en `sh fichier.sh` (sh lit le script depuis le fichier, indépendant de stdin), d'où le fait que ça passait dans nos premiers tests.

**Fix** : `</dev/null` sur les sous-process externes (`run_quiet` → couvre npx + claude mcp add, + le `claude mcp list` du check idempotence). Les prompts lisent `/dev/tty`, donc l'interactivité est intacte.

**Re-testé en condition pipe** (`cat install.sh | sh`, équivalent exact de `curl | sh`) :
- Chemin `y` + API key masquée → MCP enregistré, user scope, Connected ✅
- Chemin `n` → pitch + register link + skip propre ✅

→ Cette fois le parcours de prod est couvert pour de vrai. Prêt à merger.

**Comment by R1LGM** [comment]
Dernier polish défensif ajouté (commit `3869ffd`)

**Trap de restauration du terminal** sur la saisie masquée de l'API key. Si l'user fait `Ctrl+C` pile pendant le `Paste your LGM API key (hidden):` (entre le `stty -echo` et la restauration), son terminal restait en mode invisible (plus d'écho) jusqu'à un `reset`. Le trap (`INT`/`TERM`) réactive l'écho et exit proprement (code 130) avant de mourir.

5 lignes, zéro impact sur le happy-path (re-testé : saisie normale OK, install au bout). C'est le dernier edge case qui dégradait vraiment l'UX.

### État final — couverture de test complète

| Chemin | Statut |
|---|---|
| `curl \| sh` (pipe) — chemin `y` + API key masquée | ✅ |
| `curl \| sh` (pipe) — chemin `n` → pitch + skip | ✅ |
| `--scope user` visible depuis un autre dossier | ✅ |
| Idempotence (re-run) | ✅ |
| Verbose (`LGM_INSTALL_VERBOSE=1`) | ✅ |
| Live MCP call (campagnes listées) | ✅ |
| Ctrl+C pendant la saisie → terminal restauré | ✅ |

Plus d'edge case bloquant ou dégradant connu. Prêt à merger de ton côté.
